applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_analysis.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics as KM
+import KratosMultiphysics.StructuralMechanicsApplication as SM
+import KratosMultiphysics.ContactStructuralMechanicsApplication as CSM
+
+# Importing the solvers (if available)
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.EigenSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "not imported")
+
+# Other imports
+import sys
+
+# Import the base structural analysis
+from structural_mechanics_analysis import StructuralMechanicsAnalysis as BaseClass
+
+class ContactStructuralMechanicsAnalysis(BaseClass):
+    """
+    This class is the main-script of the ContactStructuralMechanicsApplication put in a class
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, project_parameters, external_model_part=None):
+
+        # Construct the base analysis.
+        super().__init__(project_parameters, external_model_part)
+
+    #### Internal functions ####
+    def _CreateSolver(self, external_model_part=None):
+        """ Create the Solver (and create and import the ModelPart if it is not passed from outside) """
+        if external_model_part != None:
+            # This is a temporary solution until the importing of the ModelPart
+            # is removed from the solver (needed e.g. for Optimization)
+            if (type(external_model_part) != KM.ModelPart):
+                raise Exception("Input is expected to be provided as a Kratos ModelPart object")
+            self.using_external_model_part = True
+        else:
+            self.using_external_model_part = False
+
+        ## Get echo level and parallel type
+        self.echo_level = self.ProjectParameters["problem_data"]["echo_level"].GetInt()
+        self.parallel_type = self.ProjectParameters["problem_data"]["parallel_type"].GetString()
+
+        # To avoid many prints # TODO leave this?
+        if (self.echo_level == 0):
+            KM.Logger.GetDefaultOutput().SetSeverity(KM.Logger.Severity.WARNING)
```

Comment:
I would remove this (feel free to remove it also in StructuralMechanics)
This should only be used in the tests